### PR TITLE
Add a script to dumps DB (puzzles only)

### LINKED ISSUE #94: Look into dumping/uploading DB data

### PR DESCRIPTION
**Pull request by ketphan02**
Closes #94

### CODE REVIEW

**Comment by ketphan02** [comment]
Some changes were caused by running `npm run prettify`, I think we need to implement [husky](https://www.npmjs.com/package/husky), this helps format through `.prettierrc.js`

**Comment by Opeyem1a** [comment]
I think it's something to make a card about and do later - doesn't seem critical to me, but I could be misunderstanding the problem @ketphan02

File: prisma/seed.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+import { PrismaClient, Prisma } from '@prisma/client';
+import * as faker from 'faker';
+
+const prisma = new PrismaClient();
+
+const main = async () => {
```
Comment (Opeyem1a) [inline prisma/seed.ts:9]: This is much closer to what I meant, but I still think the actual hardcoded data should not be in this file, it should still be an external `.json`, is that not possible?

Like it should be:
```ts
PUZZLE_TYPES.forEach(puzzleType => {
  const puzzleType = await prisma.puzzleType.create({
    data: {
      name: puzzleType.name
    }
  });
};
```

Where `PUZZLE_TYPES` is an array of objects defined in a separate file and imported here

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -24,6 +24,10 @@ Then run the following lines in the terminal
 npm i
 npm run dev
 ```
+If you want to have some puzzle without inputing every of them manually, run:
+```bash
+npx prisma db seed
```
[the remark this answers is not in the document]
Comment (ketphan02) [reply README.md:29]: it will trigger the dump function

File: package.json [hunk 3]
```diff
@@ -34,7 +34,7 @@
   "devDependencies": {
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^12.1.1",
-    "@types/faker": "^6.6.9",
+    "@types/faker": "^5.5.9",
```
[the remark this answers is not in the document]
Comment (ketphan02) [reply package.json:37]: faker.js 's creator has stopped maintaining the package, other packages are not as good as the latest working version so i think we should stick with this until something better rises

File: package.json [hunk 4]
```diff
@@ -53,5 +53,8 @@
     "supertest": "^6.2.2",
     "ts-jest": "^27.0.6",
     "typescript": "^4.4.4"
+  },
+  "prisma": {
+    "seed": "ts-node --compiler-options {\"module\":\"CommonJS\"} prisma/seed.ts"
```
[the remark this answers is not in the document]
Comment (ketphan02) [reply package.json:58]: it helps the script use import/export instead of require ones

File: prisma/seed.ts (hunk 1 above)
Comment (Opeyem1a) [reply prisma/seed.ts:9]: Yep yep!

**Comment by Opeyem1a** [comment]
Yikers tests are failing

File: prisma/seed/index.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+dotenv.config();
+
+import { PrismaClient, Puzzle, PuzzleType } from '@prisma/client';
+import * as faker from 'faker';
+
+const prisma = new PrismaClient();
+
+import seedData from './seed.json';
+import type { DIFFICULTY } from '../../src/types/global';
+console.log(seedData);
```
Comment (Opeyem1a) [inline prisma/seed/index.ts:11]: remove

File: prisma/seed/seed.json [hunk 6]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (Opeyem1a) [inline prisma/seed/seed.json:1]: Niice, make me a trello card to update this

File: prisma/seed/index.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    );
+    await prisma.puzzleInstance.create({
+      data: {
+        puzzle: {
+          connect: {
+            id: puzzle.id
+          }
+        },
+        hint: puzzleInstance.hint,
+        longitude: parseFloat(faker.address.longitude()),
```
Comment (Opeyem1a) [inline prisma/seed/index.ts:74]: Hmm does this need to be fake? Couldn't we define these too?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: prisma/seed/index.ts (hunk 7 above)
Comment (ketphan02) [reply prisma/seed/index.ts:74]: oh you want these to be defined too, i'll do that

Comment (ketphan02) [reply prisma/seed/index.ts:74]: I thought you want this haha

**Comment by ketphan02** [comment]
The locations:
image

File: prisma/seed/index.ts (hunk 7 above)
Comment (Opeyem1a) [reply prisma/seed/index.ts]: Actually yeah I guess I'll put real data here so it doesn't really matter

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**